tests/experimental/wgit/test_signal_sparsity.py

```diff
@@ -0,0 +1,131 @@
+        assert all((default_sst == sst).flatten())
+        assert sst.shape == tensor.shape
+        return default_sst, sst
+
+    return assert_and_get_sps_sst
+
+
+def test_validate_conf():
+    """Tests the config validation for the Signal Sparsity class."""
+
+    def kwargs(vals_list):
+        """Maps the values in input vals_list to the keys in arg_key_list"""
+
+        arg_key_list = [
+            "sst_top_k_element",
+            "sst_top_k_percent",
+            "sst_top_k_dim",
+            "dst_top_k_element",
+            "dst_top_k_percent",
+            "dst_top_k_dim",
+        ]
+        return dict(zip(arg_key_list, vals_list))
+
+    # Validate assertion error is raised when, either
+    # 1. One and only one of sst (or dst) percent and element is not provided a value (not None).
+    # 2. Both of sst (or dst) percent and element is set to None.
+    # 3. top_k_percent and top_k_element are not in valid range (elem > 0) and for 0 < percent <= 100.
+    element = 10
+    percent = 50
+    dim = 0
+    args_list = [
+        [element, percent, dim, element, None, dim],  # case 1.
+        [element, None, dim, element, percent, dim],
+        [None, None, dim, element, None, dim],  # case 2.
+        [element, None, dim, None, None, dim],
+        [0, None, dim, None, None, dim],  # case 3.
+        [None, 0, dim, None, None, dim],
+        [element, None, dim, 0, None, dim],
+        [element, None, dim, None, 0, dim],
+    ]
+    for args in args_list:
+        pytest.raises(ValueError, SignalSparsity, **kwargs(args))
+
+
+def test_dense_to_sst(sst_and_default_sps):
+    """Tests the dense_to_sst method with fixed inputs"""
+    tensors = list()
+    tensors.append(torch.arange(20).reshape(4, 5))
+    tensors.append(torch.arange(80).reshape(4, 5, 4))
+
+    def elem_percent_equality(tensor, percent, element, dim):
+        """verifies if the topk element and top k percent returns the same results
+        for same sparsity in a fixed tensor along some dim.
+        """
+        _, sst_percent = sst_and_default_sps(tensor, percent, None, dim=dim)
+        _, sst_elem = sst_and_default_sps(tensor, None, element, dim=dim)
+        assert all((sst_percent == sst_elem).flatten())
+
+    for tensor in tensors:
+        dim_list = [None] + list(range(tensor.dim()))
+        for dim in dim_list:
+            # Should return the FFT transformed tensor with top_100_percent for sst
+            sparser_2d = SignalSparsity(sst_top_k_percent=100, sst_top_k_dim=dim, dst_top_k_percent=100)
+            assert all((sparser_2d.dense_to_sst(tensor) == torch.fft.fft(tensor)).flatten())
+
+            # Test for different sparsity levels using top_k_percent and top_k_elements
+            for percent, element in zip([20, 40, 60, 80], [1, 2, 3, 4]):
+                sst_and_default_sps(tensor, percent=percent, element=None, dim=dim)
+                sst_and_default_sps(tensor, percent=None, element=element, dim=dim)
```

Comment:
do we need to test those different values? I would say test only a common case + corner cases.